packages/components/src/autocomplete/types.ts

```diff
@@ -0,0 +1,214 @@
+/**
+ * WordPress dependencies
+ */
+import type { WPElement } from '@wordpress/element';
+/**
+ * External dependencies
+ */
+import type { MutableRefObject, ReactNode } from 'react';
+/**
+ * Internal dependencies
+ */
+import type { useAutocomplete } from '.';
+
+type GetOptions< TCompleterOption = any > = (
+	filterValue: AutocompleterUIProps[ 'filterValue' ]
+) => Array< TCompleterOption > | Promise< Array< TCompleterOption > >;
+
+export type OptionCompletion = {
+	action: 'insert-at-caret' | 'replace';
+	value: string | WPElement;
+};
+
+type getOptionCompletion< TCompleterOption = any > = (
```

Comment:
We probably want to PascalCase all types for convention:

```suggestion
type GetOptionCompletion< TCompleterOption = any > = (
```

(Or alternatively, this one could also just be inlined instead of using a type alias.)

Reply:
gah, nice catch on the casing, thank you.

I do like the inline suggestion. Simplifies the file and also lets this prop use the same generic as the rest of the `WPCompelter` type instead of declaring its own 👍 14351698d7